### PR TITLE
added tests for PG and SF schema changes, fixed edge cases and quoting

### PR DESCRIPTION
**Pull request by heavycrystal**
This PR adds on to the existing work towards syncing schema changes in CDC mirrors (#368):

1. handles an edge case where a column is dropped and recreated with a different data type in the same ALTER TABLE statement.
2. Adds quoting, guarding against case-sensitive column names and reserved identifiers.
3. Fixes type mapping for BigQuery

Additionally, adds some tests for ReplayTableSchemaDelta for Postgres and Snowflake. Bigquery does not have tests yet but has been manually tested.

### CODE REVIEW

File: flow/connectors/postgres/cdc.go [hunk 1]
```diff
@@ -545,13 +545,13 @@
...
-	currRelMap := make(map[string]bool)
+	prevRelMap := make(map[string]*uint32)
+	currRelMap := make(map[string]*uint32)
 	for _, column := range prevRel.Columns {
-		prevRelMap[column.Name] = true
+		prevRelMap[column.Name] = &column.DataType
 	}
 	for _, column := range currRel.Columns {
-		currRelMap[column.Name] = true
+		currRelMap[column.Name] = &column.DataType
```
[the remark this answers is not in the document]
Comment (heavycrystal) [reply flow/connectors/postgres/cdc.go]: fixed

File: flow/connectors/postgres/cdc.go [hunk 2]
```diff
@@ -545,13 +545,13 @@ func (p *PostgresCDCSource) processRelationMessage(
 	// retrieve initial RelationMessage for table changed.
 	prevRel := p.relationMessageMapping[currRel.RelationId]
 	// creating maps for lookup later
-	prevRelMap := make(map[string]bool)
-	currRelMap := make(map[string]bool)
+	prevRelMap := make(map[string]*uint32)
+	currRelMap := make(map[string]*uint32)
 	for _, column := range prevRel.Columns {
-		prevRelMap[column.Name] = true
+		prevRelMap[column.Name] = &column.DataType
```
[the remark this answers is not in the document]
Comment (heavycrystal) [reply flow/connectors/postgres/cdc.go]: fixed
